hazelcast/src/main/java/com/hazelcast/instance/Node.java

```diff
@@ -479,14 +479,17 @@ private void waitIfAlreadyShuttingDown() {
             return;
         }
         logger.info("Node is already shutting down... Waiting for shutdown process to complete...");
-        while (state != NodeState.SHUT_DOWN) {
+        while (state != NodeState.SHUT_DOWN && shuttingDown.get()) {
```

Comment:
dumb question: I guess you need the 'shutting down' field because 'state' has no mode for 'shutting down'; only 'shutdown' right?